src/Workspaces/CSharp/Portable/LanguageServices/CSharpSemanticFactsService.cs

```diff
@@ -231,6 +231,18 @@ public ForEachSymbols GetForEachSymbols(SemanticModel semanticModel, SyntaxNode
             }
         }
 
+        public IMethodSymbol GetAwaitExpressionMethod(SemanticModel semanticModel, SyntaxNode node)
```

Comment:
as weird as it sounds, i think this should be: GetGetAwaiterMethod, or if hat sounds awful: GetInvokedGetAwaiterMethod.

Reply:
Is AwaitExpressionInfo shared across languages?  If so, this could also be: GetAwaitExpressionInfo.

Reply:
Unfortunately, there are two `AwaitExpressionInfo` types (one for VB and one for C#).

Reply:
Both `AwaitExpressionExpressionInfo` types are structs :-(
So if we want a common experience for callers, we'd have to use some interface.

Reply:
Or a Common-Struct.  we do that for some other APIs.